lib/spack/llnl/util/lang.py

```diff
@@ -174,22 +175,44 @@ def union_dicts(*dicts):
     return result
 
 
+# Used as a sentinel that disambiguates tuples passed in *args from coincidentally
+# matching tuples formed from kwargs item pairs.
+_kwargs_separator = (object(),)
+
+
+def equal_args(*args, **kwargs):
```

Comment:
This function doesn't actually check equality of anything. I think a more appropriate name would be `args_repr`, `stable_args`, or something else to that effect.

Reply:
Totally right! Done in 50c3b21.